utils/utils.go

```diff
@@ -302,3 +302,16 @@ func MoveFileIfExists(source, dest string) error {
 	}
 	return nil
 }
+
+func NewPullSecretFromDockerConfig(dockerConfigJSON []byte) *corev1.Secret {
```

Comment:
why not to add 
`if err := common.RetryOnConflictOrRetriable(retry.DefaultBackoff, func() error {
		return c.Update(ctx, originalPullSecret)
	}); err != nil {
		return fmt.Errorf("failed to restore original pull-secret in seed cluster: %w", err)
	}`
here? It seems common to all places no?